OmiseSwift/Listable.swift

```diff
@@ -50,4 +52,105 @@ public extension Listable where Self: ResourceObject {
         let client = resolveClient(given: given)
         return client.call(operation, callback: callback)
     }
+    
+    @discardableResult
+    public static func list(using given: Client? = nil, parent: ResourceObject? = nil, listParams: ListParams? = nil, callback: @escaping (Failable<List<Self>>) -> Void) -> Request<ListOperation.Result>? {
+        guard checkParent(withContext: self, parent: parent) else {
+            return nil
+        }
+        
+        let operation = self.listOperation(parent, params: listParams)
+        let client = resolveClient(given: given)
+        
+        let requestCallback: ListOperation.Callback = { result in
+            switch result {
+            case .success(let result):
+                let list = List(endpoint: operation.endpoint, paths: operation.pathComponents, order: listParams?.order, list: result)
+                callback(.success(list))
+            case .fail(let error):
+                callback(.fail(error))
+            }
+        }
+        
+        return client.call(operation, callback: requestCallback)
+    }
```

Comment:
I think we can use that. This closure will call the given callback not to convert it though